python/tvm/relay/frontend/tensorflow.py

```diff
@@ -1458,6 +1458,15 @@ def _impl(inputs, attr, params, mod):
 
                 return ret
 
+        def _dyn():
+            for d in data_shape:
+                if not isinstance(d, int):
+                    return True
+            return False
+
+        if _dyn():
```

Comment:
Why do we need to special hand this in tf frontend and skip mask transformation?

Reply:
Because mask transformation needs a concrete shape which doesn't work for dynamic shape input. I think there is more to do to handle mask in dynamic case but I'm afraid it can't support all mask transformation without concrete shape.

Reply:
We can use shape_of to get data shape. In this case all shape dim will be relay expression and transform mask should be able to handle them. Anyway I don't think we should silently skip it since the output can be wrong and cause latter type infer error.

Reply:
@lixiaoquan Can you simply raise an error here and add a TODO? We can merge this PR so that backend changes can take effect.